python/lsst/afw/image/testUtils.py

```diff
@@ -36,125 +41,108 @@ def makeGaussianNoiseMaskedImage(dimensions, sigma, variance=1.0):
-    """Compare pixels from two masked images
-    
-    Inputs:
-    - maskedImageArrSet1: first masked image to compare as (image, mask, variance) arrays
-    - maskedImageArrSet2: second masked image to compare as (image, mask, variance) arrays
-    - doImage: compare image planes if True
-    - doMask: compare mask planes if True
-    - doVariance: compare variance planes if True
-    - skipMaskArr: pixels to ingore on the image, mask and variance arrays; nonzero values are skipped
-    - rtol: relative tolerance (see below)
-    - atol: absolute tolerance (see below)
+@lsst.utils.tests.inTestCase
+def assertMaskedImagesNearlyEqual(testCase, maskedImage0, maskedImage1,
+    doImage=True, doMask=True, doVariance=True, skipMask=None,
+    rtol=1.0e-05, atol=1e-08, msg="Masked images differ"):
+    """!Assert that two masked images are nearly equal, including non-finite values
     
-    rtol and atol are positive, typically very small numbers.
-    The relative difference (rtol * abs(b)) and the absolute difference "atol" are added together
-    to compare against the absolute difference between "a" and "b".
+    @param[in] testCase  unittest.TestCase instance the test is part of;
+                        an object supporting one method: fail(self, msgStr)
+    @param[in] maskedImage0  masked image 0 (an lsst.afw.image.MaskedImage or
+        collection of three transposed numpy arrays: image, mask, variance)
+    @param[in] maskedImage1  masked image 1 (an lsst.afw.image.MaskedImage or
+        collection of three transposed numpy arrays: image, mask, variance)
+    @param[in] doImage  compare image planes if True
+    @param[in] doMask  compare mask planes if True
+    @param[in] doVariance  compare variance planes if True
+    @param[in] skipMask  mask of pixels to skip, or None to compare all pixels;
+        an lsst.afw.image.Mask, lsst.afw.image.Image, or transposed numpy array;
+        all non-zero pixels are skipped
+    @param[in] rtol  maximum allowed relative tolerance; more info below
+    @param[in] atol  maximum allowed absolute tolerance; more info below
+    @param[in] msg  exception message prefix; details of the error are appended after ": "
+
+    The mask planes must match exactly. The image and variance planes are nearly equal if all pixels obey:
+        |val1 - val0| <= rtol*|val1| + atol
+    or, for float types, if nan/inf/-inf pixels match.
+
+    @warning the comparison equation is not symmetric, so in rare cases the assertion
+    may give different results depending on which masked image comes first.
     
     Return a string describing the error if the images differ significantly, an empty string otherwise
     """
-    retStrs = []
+    maskedImageArrSet0 = maskedImage0.getArrays() if hasattr(maskedImage0, "getArrays") else maskedImage0
+    maskedImageArrSet1 = maskedImage1.getArrays() if hasattr(maskedImage1, "getArrays") else maskedImage1
+
+    try:
+        assert len(maskedImageArrSet0) == 3
```

Comment:
These sanity checks don't seem to be very bullet proof. If I pass in any sequence with 3 elements in it (including a 3 character long string) I will get through all this code.

If we really want to check there are 2 x 3 transposed numpy arrays that are the same shape then we should check for that. If the intention is purely to ensure we have three whatevers to pass to `imagesDiffer` then we may want to note that here and also modify the error message. You aren't checking whether it's a supported type, only that you have 3 of them.